src/EditorFeatures/Core/Extensibility/Completion/CompletionHelper.cs

```diff
@@ -139,52 +139,37 @@ protected static bool IsAllDigits(string filterText)
             return null;
         }
 
-        /// <summary>
-        /// Apply any culture-specific quirks to the given text for the purposes of pattern matching.
-        /// For example, in the Turkish locale, capital 'i's should be treated specially in Visual Basic.
-        /// </summary>
-        protected virtual string GetCultureSpecificQuirks(string candidate)
-        {
-            return candidate;
-        }
-
         private readonly object _gate = new object();
         private readonly Dictionary<string, PatternMatcher> _patternMatcherMap = new Dictionary<string, PatternMatcher>();
         private readonly Dictionary<string, PatternMatcher> _fallbackPatternMatcherMap = new Dictionary<string, PatternMatcher>();
         internal static readonly CultureInfo EnUSCultureInfo = new CultureInfo("en-US");
 
-        protected PatternMatcher GetPatternMatcher(string value, CultureInfo culture)
+        private static PatternMatcher GetPatternMatcher(
+            object gate, string value, CultureInfo culture, Dictionary<string, PatternMatcher> map)
```

Comment:
Passing the object to lock on seems strange. Consider dropping the `static` and locking on the field again?